draft-ietf-quic-transport.md

```diff
@@ -1834,9 +1834,18 @@ SHOULD include information that allows the server to verify that the source IP
 address and port in client packets remains constant.
 
 Servers might use tokens from NEW_TOKEN in deciding not to send a Retry packet,
-even if the client address has changed. Tokens sent in NEW_TOKEN frames SHOULD
-include information that allows the server to verify if the client address is
-stable, but might allow for different NAT bindings or ephemeral port selection.
+even if the client address has changed. Tokens sent in NEW_TOKEN frames MUST
+include information that allows the server to verify that the client IP address
+has not changed from when the token was issued.
```

Comment:
I think this text would be a lot clearer if you moved the first sentence to the end and wrote it as:

"Servers might use tokens from NEW_TOKEN in deciding not to send a Retry packet,
even if the client address has changed. However, if the address has changed, the
server MUST adhere to the anti-amplification limits found in XXX"

Reply:
That works.  I think that is better, though I want to be more precise about "address" vs. "IP address" here.  "Address" in this doc refers to IP+port, and I think that we do want to allow clients to choose different ephemeral ports or to be behind a NAT without absolutely requiring servers to use address validation.  I know that adds a bit of uncertainty to this, but it's a compromise.

I've replaced "address" with "IP address" and then replaced the long note with a shorter sentence explaining that IP alone is not perfect.  I think that we might need wider discussion on this one point.